### DISCUSSION TITLE
MutationCache undocumented methods

### DISCUSSION THREAD

**Discussion by Birkbjo** [TanStack/query#4181] [Q&A] [answered]
Hi!

I've a simple question: there's multiple methods on `mutationCache` that are not  [documented here](https://tanstack.com/query/v4/docs/reference/MutationCache), eg. `remove()`, `find()`, `findAll()`. Are these safe to use, or are they not considered part of the public API?

**Comment by TkDodo** [comment, marked as answer]
I wouldn't rely on it. For example, `find` is totally unused in our code-base, so I might remove it as it's undocumented. Generally, we try to respect public methods though.

**Reply by Birkbjo** [reply]
So is there another way to do something like this: https://github.com/TanStack/query/discussions/4131 (deduplicate identical mutations). 
How would I remove a particular mutation while I'm offline? Is there another way to do it, or is the consensus to just send all mutations to the server to more accurately simulate online behaviour?
